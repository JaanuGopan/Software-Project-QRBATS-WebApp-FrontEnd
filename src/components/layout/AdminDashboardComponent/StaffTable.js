import React, { useState, useEffect } from "react";
import "../../../pages/AdminDashboard/AdminDashboard.css";
import { FaEdit } from "react-icons/fa";

const StaffTable = ({
  search,
  handleUpdateStaff,
  onStaffClick,
  staffsList,
}) => {
  const [staffs, setStaffs] = useState([]);
  const [selectedStaff, setSelectedStaff] = useState(null);

  useEffect(() => {
    // Fetch the list of staffs from the API
    setStaffs(staffsList);
  }, [staffsList]);

  const handleStaffClick = (staff) => {
    setSelectedStaff(staff);
    onStaffClick(staff);
  };

  const deparmentList = ["DEIE", "DCOM", "DMME", "DCEE", "DMENA"];

  return (
    <div className="tableDesign">
      <table className="student-tableArrangement">
        <thead>
          <tr>
            <th>No</th>
            <th className="expand">Name</th>
            <th>Department</th>
            <th>Username</th>
            <th>Password</th>
            <th>Edit</th>
          </tr>
        </thead>
        <tbody>
          {staffs
            .filter(
              (staff) =>
                staff.firstName.toLowerCase().includes(search.toLowerCase()) ||
                staff.lastName.toLowerCase().includes(search.toLowerCase()) ||
                staff.username.toLowerCase().includes(search.toLowerCase())
            )
            .map((staff, index) => (
              <tr
                key={index}
                onClick={() => handleStaffClick(staff)}
                className={selectedStaff === staff ? "selected-row" : ""}
              >
                <td>{index + 1}</td>
                <td>{staff.firstName + " " + staff.lastName}</td>
                <td>{deparmentList[staff.departmentId - 1]}</td>
                <td>{staff.username}</td>
                <td>********</td>
                <td>
                  <button
                    onClick={() => handleUpdateStaff(staff)}
                    className="EditButton"
                  >
                    <FaEdit className="EditIcon" />
                  </button>
                </td>
              </tr>
            ))}
        </tbody>
      </table>
    </div>
  );
};

export default StaffTable;
