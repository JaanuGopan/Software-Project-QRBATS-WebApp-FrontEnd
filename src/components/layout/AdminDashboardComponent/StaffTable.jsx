import React, { useState, useEffect } from 'react';
import '../../../pages/AdminDashboard/AdminDashboard.css';
import { FaEdit } from 'react-icons/fa';
import Department from '../../../utils/Department';

const StaffTable = ({ search, handleUpdateStaff, onStaffClick, staffsList }) => {
  const [selectedStaff, setSelectedStaff] = useState(null);

  const handleStaffClick = (staff) => {
    setSelectedStaff(staff);
    onStaffClick(staff);
  };

  const departmentList = Department.departmentList;

  return (
    <div className="tableDesign">
      <table className="student-tableArrangement">
        <thead>
          <tr>
            <th>No</th>
            <th className="expand">Name</th>
            <th>Department</th>
            <th>Email</th>
            <th>Edit</th>
          </tr>
        </thead>
        {staffsList
            .filter(
              (staff) =>
                staff.firstName.toLowerCase().includes(search.toLowerCase()) ||
                staff.lastName.toLowerCase().includes(search.toLowerCase()) ||
                staff.username.toLowerCase().includes(search.toLowerCase())
            ).length > 0 ? <tbody>
          {staffsList
            .filter(
              (staff) =>
                staff.firstName.toLowerCase().includes(search.toLowerCase()) ||
                staff.lastName.toLowerCase().includes(search.toLowerCase()) ||
                staff.username.toLowerCase().includes(search.toLowerCase())
            )
            .map((staff, index) => (
              <tr
                key={index}
                onClick={() => handleStaffClick(staff)}
                className={selectedStaff === staff ? 'selected-row' : ''}
              >
                <td>{index + 1}</td>
                <td>{staff.firstName + ' ' + staff.lastName}</td>
                <td>{departmentList[staff.departmentId - 1]}</td>
                <td>{staff.email}</td>

                <td>
                  <button onClick={() => handleUpdateStaff(staff)} className="EditButton">
                    <FaEdit className="EditIcon" />
                  </button>
                </td>
              </tr>
            ))}
        </tbody> : <tbody className='nodata'>
          No Data Available
        </tbody>}
      </table>
    </div>
  );
};

export default StaffTable;
