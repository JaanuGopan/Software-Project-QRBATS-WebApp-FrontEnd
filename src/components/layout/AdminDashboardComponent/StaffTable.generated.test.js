import renderer from 'react-test-renderer';
import React, { useState, useEffect } from "react";

import { FaEdit } from "react-icons/fa";
import StaffTable from "./StaffTable";

jest.mock("../../../pages/AdminDashboard/AdminDashboard.css");
jest.mock("react-icons/fa");

const renderTree = tree => renderer.create(tree);
describe('<StaffTable>', () => {
  it('should render component', () => {
    expect(renderTree(<StaffTable 
    />).toJSON()).toMatchSnapshot();
  });
  
});