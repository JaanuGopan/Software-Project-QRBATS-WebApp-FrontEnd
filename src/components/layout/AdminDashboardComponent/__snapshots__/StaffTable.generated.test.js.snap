// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StaffTable> should render component 1`] = `
<div
  className="tableDesign"
>
  <table
    className="student-tableArrangement"
  >
    <thead>
      <tr>
        <th>
          No
        </th>
        <th
          className="expand"
        >
          Name
        </th>
        <th>
          Department
        </th>
        <th>
          Username
        </th>
        <th>
          Password
        </th>
        <th>
          Edit
        </th>
      </tr>
    </thead>
    <tbody />
  </table>
</div>
`;
